### ISSUE TITLE
java.lang.StackOverflowError

### ISSUE THREAD

**Issue by mdsb** [opt-tech/redshift-fake-driver#18]
When using redshift-fake-driver library with postgresql (PostgreSQL JDBC Driver) maven dependency, i get java.lang.StackOverflowError. 
Could you please help.

**Comment by ocadaruma** [comment]
@mdsb Could you give me some information to reproduce ?

- Driver version
- PostgreSQL version
- Code snippet (if you can)

**Comment by mdsb** [comment]
@ocadaruma  - Thank you for asking. I have resolved this Stackoverflow error.
However, I was hoping your library would help in unit testing the com.databricks.spark-redshift_2.11.
If possible, please let me know , how to unit test the code that uses com.databricks.spark-redshift_2.11.

Following is the snippet that I'm hoping to test.
`dataFrameToPersist
      .write.format("com.databricks.spark.redshift")
      .options(redshiftDefaultOptionsMap)
      .option("dbtable", finalRedshiftTable.replaceAll("-","_"))
      .mode(SaveMode.Overwrite)
      .save()`

**Comment by IvanAtanasov** [comment]
Did you ever solve this @ocadaruma ?

Looking to do the same myself and getting a stackoverflow error. Guessing after I resolve that I am going to run into more problems.

**Comment by ocadaruma** [comment]
Since spark-redshift does not support fake s3, it seems real S3 bucket is needed for unit testing.
I will investigate further when I have time.

**Comment by mpeschke-livongo** [comment]
I'm also having this issue. Throwing a shot in the dark, specifying the driver in the properties (per the example in the readme) might cause it to resolve back to the {{FakePostgresqlDriver}}.

Here is the code triggering the `StackOverflowError`:
```scala
    Class.forName("jp.ne.opt.redshiftfake.postgres.FakePostgresqlDriver")
    val properties = new Properties
    properties.setProperty("username", "profiler-test")
    properties.setProperty("password", "1234")
    properties.setProperty("driver", "jp.ne.opt.redshiftfake.postgres.FakePostgresqlDriver")
    val conn = DriverManager.getConnection("jdbc:postgresqlredshift://127.0.0.1:8432", properties)
```

Here's the top of the stack:
```
java.lang.StackOverflowError
	at sun.net.spi.DefaultProxySelector$3.run(DefaultProxySelector.java:218)
	at sun.net.spi.DefaultProxySelector$3.run(DefaultProxySelector.java:208)
	at java.security.AccessController.doPrivileged(Native Method)
	at sun.net.spi.DefaultProxySelector.select(DefaultProxySelector.java:207)
	at java.net.SocksSocketImpl.connect(SocksSocketImpl.java:384)
	at java.net.Socket.connect(Socket.java:589)
	at java.net.Socket.connect(Socket.java:538)
	at java.net.Socket.<init>(Socket.java:434)
	at java.net.Socket.<init>(Socket.java:211)
	at org.postgresql.core.PGStream.<init>(PGStream.java:62)
	at org.postgresql.core.v3.ConnectionFactoryImpl.openConnectionImpl(ConnectionFactoryImpl.java:76)
	at org.postgresql.core.ConnectionFactory.openConnection(ConnectionFactory.java:66)
	at org.postgresql.jdbc2.AbstractJdbc2Connection.<init>(AbstractJdbc2Connection.java:125)
	at org.postgresql.jdbc3.AbstractJdbc3Connection.<init>(AbstractJdbc3Connection.java:30)
	at org.postgresql.jdbc3g.AbstractJdbc3gConnection.<init>(AbstractJdbc3gConnection.java:22)
	at org.postgresql.jdbc4.AbstractJdbc4Connection.<init>(AbstractJdbc4Connection.java:30)
	at org.postgresql.jdbc4.Jdbc4Connection.<init>(Jdbc4Connection.java:24)
	at org.postgresql.Driver.makeConnection(Driver.java:393)
	at org.postgresql.Driver.connect(Driver.java:267)
	at java.sql.DriverManager.getConnection(DriverManager.java:664)
	at java.sql.DriverManager.getConnection(DriverManager.java:208)
	at jp.ne.opt.redshiftfake.postgres.FakePostgresqlDriver.connect(FakePostgresqlDriver.java:27)
	at java.sql.DriverManager.getConnection(DriverManager.java:664)
	at java.sql.DriverManager.getConnection(DriverManager.java:208)
	at jp.ne.opt.redshiftfake.postgres.FakePostgresqlDriver.connect(FakePostgresqlDriver.java:27)
	at java.sql.DriverManager.getConnection(DriverManager.java:664)
	at java.sql.DriverManager.getConnection(DriverManager.java:208)
	at jp.ne.opt.redshiftfake.postgres.FakePostgresqlDriver.connect(FakePostgresqlDriver.java:27)
	at java.sql.DriverManager.getConnection(DriverManager.java:664)
	at java.sql.DriverManager.getConnection(DriverManager.java:208)
	at jp.ne.opt.redshiftfake.postgres.FakePostgresqlDriver.connect(FakePostgresqlDriver.java:27)
	at java.sql.DriverManager.getConnection(DriverManager.java:664)
	at java.sql.DriverManager.getConnection(DriverManager.java:208)
	at jp.ne.opt.redshiftfake.postgres.FakePostgresqlDriver.connect(FakePostgresqlDriver.java:27)
	at java.sql.DriverManager.getConnection(DriverManager.java:664)
	at java.sql.DriverManager.getConnection(DriverManager.java:208)
        ...
```

**Comment by ocadaruma** [comment]
I figured out the problem.
Postgres's drvier does not accepts `jdbc:postgresql://127.0.0.1:8432`, because the URL does not contain database name.
If you specify `jdbc:postgresqlredshift://127.0.0.1:8432/foo`, stack over flow does not occurs.

And, I fixed driver behavior in #29 .
The driver no more causes StackOverflowError (but cause SQLException instead)
